api/genericresource/helpers.go

```diff
@@ -0,0 +1,48 @@
+package genericresource
+
+import (
+	"fmt"
+
+	"github.com/docker/swarmkit/api"
+)
+
+// NewError creates a new error for Third Party Resources
+func NewError(format string, args ...interface{}) error {
+	return fmt.Errorf("Could not parse GenericResource: "+format, args...)
+}
+
+// NewSet creates a set object
+func NewSet(keys ...string) *api.GenericResource_Set {
+	keyMap := make(map[string]*api.Set_Void)
+
+	for _, k := range keys {
+		keyMap[k] = nil
+	}
+
+	return &api.GenericResource_Set{Set: &api.Set{Val: keyMap}}
+}
+
+// NewInteger creates an Integer object
+func NewInteger(val uint64) *api.GenericResource_Integer {
```

Comment:
I think Discrete is better than Integer. This type of resource has to be assigned in unit.